### PR TITLE
Refactor balloon to C++

### PR DESCRIPTION
**Pull request by IntelOrca**

### CODE REVIEW

File: src/openrct2/world/Balloon.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    }
+
+    void balloon_update(rct_balloon * balloon)
+    {
+        balloon->Update();
+    }
+
+    void game_command_balloon_press(sint32 * eax, sint32 * ebx, sint32 * ecx, sint32 * edx, sint32 * esi, sint32 * edi, sint32 * ebp)
+    {
+        uint16 spriteIndex = *eax & 0xFFFF;
```
Comment (duncanspumpkin) [inline src/openrct2/world/Balloon.cpp]: I find it much nicer to read when the game command is just a stub and it calls a separate function. That way its much easier to know what registers are passed into and out of a command.

Comment (IntelOrca) [inline src/openrct2/world/Balloon.cpp]: I did that originally then changed it back.

File: src/openrct2/world/Balloon.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        }
+        return 0;
+    }
+}
+
+extern "C"
+{
+    void create_balloon(sint32 x, sint32 y, sint32 z, sint32 colour, bool isPopped)
+    {
+        rct_sprite* sprite = create_sprite(2);
```
Comment (duncanspumpkin) [inline src/openrct2/world/Balloon.cpp]: `SPRITE_IDENTIFIER_MISC` although im not completely convinced that this is the correct thing to pass to `create_sprite`

File: src/openrct2/world/sprite.h [hunk 3]
```diff
@@ -337,6 +344,12 @@ typedef union {
 	rct_crashed_vehicle_particle crashed_vehicle_particle;
 	rct_crash_splash crash_splash;
 	rct_steam_particle steam_particle;
+
+#ifdef __cplusplus
+    bool IsBalloon();
```
Comment (duncanspumpkin) [inline src/openrct2/world/sprite.h]: tabs

File: src/openrct2/world/Balloon.cpp (hunk 2 above)
Comment (IntelOrca) [inline src/openrct2/world/Balloon.cpp]: This parameter is a set of flags, but there is only one flag (2nd bit) which says whether it goes in the misc list or not.

**Pull request merged**